module/zfs/zio.c

```diff
@@ -1159,9 +1159,9 @@ zio_flush(zio_t *zio, vdev_t *vd)
 void
 zio_shrink(zio_t *zio, uint64_t size)
 {
-	ASSERT(zio->io_executor == NULL);
-	ASSERT(zio->io_orig_size == zio->io_size);
-	ASSERT(size <= zio->io_size);
+	ASSERT3P(zio->io_executor, ==, NULL);
+	ASSERT3P(zio->io_orig_size, ==, zio->io_size);
```

Comment:
This should be `ASSERT3U`.  The build failure you're seeing is coming from 32-bit system where the pointer size is 32-bit and not 64-bit.  The ASSERT3P macro is forcing a cast of a `uint64_t` to a `uintptr_t` which is only 32-bit on failing architectures. It looks like ASSERT3P is used incorrectly in a zio_wait() too and maybe elsewhere.